akka-projection-grpc-tests/src/it/scala/akka/persistence/typed/EventWriterSpec.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed
+
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.pattern.StatusReply
+import com.typesafe.config.ConfigFactory
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+object EventWriterSpec {
+  def config =
+    ConfigFactory.parseString(
+      // until we have it in reference.conf
+      """
+      akka.persistence.typed.event-writer {
+        max-batch-size = 10
+        ask-timeout = 5s
+      }
+      """)
+}
+
+class EventWriterSpec extends ScalaTestWithActorTestKit(EventWriterSpec.config) with AnyWordSpecLike with LogCapturing {
+
+  implicit val ec: ExecutionContext = testKit.system.executionContext
+
+  "The event writer" should {
+
+    "handle duplicates" in {
+      val writer = spawn(EventWriter("akka.persistence.journal.inmem"))
+
+      val probe = createTestProbe[StatusReply[EventWriter.WriteAck]]()
+      writer ! EventWriter.Write("pid1", 1L, "one", None, Set.empty, probe.ref)
+      probe.receiveMessage().getValue
+
+      // should also be ack:ed
+      writer ! EventWriter.Write("pid1", 1L, "one", None, Set.empty, probe.ref)
+      probe.receiveMessage().getValue
+    }
+
+    "handle batched duplicates" in {
+      // FIXME we can't really know inmem isn't fast enough to insert each one by one, over on the Akka side we
+      //       have SteppingInMemJournal that we could perhaps use for deterministic tests, or add a delay config
+      //       to the inmem to easier simulate actual db
+      val writer = spawn(EventWriter("akka.persistence.journal.inmem"))
```

Comment:
or, we can have an alternative constructor with `journal: ActorRef` parameter. Then we can inject a test probe and reply to the journal protocol requests from the test

Reply:
Neat idea, that would make it a lot easier and more deterministic.